htdocs/core/modules/facture/mod_facture_mercure.php

```diff
@@ -145,15 +145,46 @@ function getNextValue($objsoc, $invoice, $mode='next')
         $mask = '';
         if (is_object($invoice) && $invoice->type == 1)
         {
+			$constant = 'FACTURE_MERCURE_MASK_REPLACEMENT_'.$invoice->entity;
+			if (! empty($conf->global->$constant)) {
+				$mask = $conf->global->$constant; // for multicompany proposal sharing
+			} else {
         	$mask=$conf->global->FACTURE_MERCURE_MASK_REPLACEMENT;
+			}
         	if (! $mask)
         	{
+				$constant = 'FACTURE_MERCURE_MASK_INVOICE_'.$invoice->entity;
+				if (! empty($conf->global->$constant)) {
+					$mask = $conf->global->$constant; // for multicompany proposal sharing
+				} else {
         		$mask=$conf->global->FACTURE_MERCURE_MASK_INVOICE;
         	}
         }
-        else if (is_object($invoice) && $invoice->type == 2) $mask=$conf->global->FACTURE_MERCURE_MASK_CREDIT;
-        else if (is_object($invoice) && $invoice->type == 3) $mask=$conf->global->FACTURE_MERCURE_MASK_DEPOSIT;
-        else $mask=$conf->global->FACTURE_MERCURE_MASK_INVOICE;
+        }
+        else if (is_object($invoice) && $invoice->type == 2){
+			$constant = 'FACTURE_MERCURE_MASK_CREDIT_'.$invoice->entity;
+			if (! empty($conf->global->$constant)) {
+				$mask = $conf->global->$constant; // for multicompany proposal sharing
+			} else {
+				$mask=$conf->global->FACTURE_MERCURE_MASK_CREDIT;
+			}
+		}
+		else if (is_object($invoice) && $invoice->type == 3){
+			$constant = 'FACTURE_MERCURE_MASK_DEPOSIT_'.$invoice->entity;
+			if (! empty($conf->global->$constant)) {
+				$mask = $conf->global->$constant; // for multicompany proposal sharing
+			} else {
+				$mask=$conf->global->FACTURE_MERCURE_MASK_DEPOSIT;
+			}
+		}
+        else {
+			$constant = 'FACTURE_MERCURE_MASK_INVOICE_'.$invoice->entity;
+			if (! empty($conf->global->$constant)) {
+				$mask = $conf->global->$constant; // for multicompany proposal sharing
+			} else {
+				$mask=$conf->global->FACTURE_MERCURE_MASK_INVOICE;
+			}
+		}
```

Comment:
@atm-quentin 
can you revert this, this manage by the multicompany module

Reply:
sure